### PR TITLE
Migrate from PCRE to PCRE2

### PR DESCRIPTION
**Pull request by heitbaum**
Update to using PCRE2 as PCRE is obsolete and being removed from distributions. 
Needs to be run tested by a vdr-plugin-fixer regex user.

- ref: https://github.com/PCRE2Project/pcre2/issues/51

### CODE REVIEW

File: regexp.c [hunk 1]
```diff
@@ -54,30 +45,22 @@ cRegexp::~cRegexp(void)
...
-  const char *error;
-  int erroffset;
-  re = pcre_compile(regexp, modifiers, &error, &erroffset, NULL);
-  if (error) {
-     error("PCRE compile error: %s at offset %i", error, erroffset);
+  PCRE2_UCHAR error[256];
+  PCRE2_SIZE erroffset;
+  int errnumber;
+  re = pcre2_compile((PCRE2_SPTR8)regexp, PCRE2_ZERO_TERMINATED, modifiers, &errnumber, &erroffset, NULL);
+  if (errnumber) {
```
Comment (mlehtima) [inline regexp.c]: This for some reason needs to be "if (!re)" for the code to work. Without the change I got "ERROR: PCRE compile error: no error at offset 0"

File: regexp.c [hunk 2]
```diff
@@ -54,30 +45,22 @@ cRegexp::~cRegexp(void)
...
-              enabled = false;
-              }
-           else {
-              csd = pcre_study(cre, PCRE_STUDY_JIT_COMPILE, (const char **)&error);
-              if (error)
-                 error("PCRE study error: %s", error);
-              }
+     if (cregexp) {
+        cre = pcre2_compile((PCRE2_SPTR8)cregexp, PCRE2_ZERO_TERMINATED, cmodifiers, &errnumber, &erroffset, NULL);
+        if (errnumber) {
```
Comment (mlehtima) [inline regexp.c]: This for some reason needs to be "if (!cre)" for the code to work.

**Review by mlehtima — CHANGES_REQUESTED** [review]
Thank you for your contribution. I found some issues while testing this on my VDR instance. After the fixes I mentioned in the inline comments regexps seem to be working correctly.

File: regexp.c (hunk 1 above)
Comment (heitbaum) [reply regexp.c]: Guess a lack of my understanding / reading on errnumber. But !re/cre achieves the same. 👍

File: regexp.c (hunk 2 above)
Comment (heitbaum) [reply regexp.c]: Done

**Comment by heitbaum** [comment]
> Thank you for your contribution. I found some issues while testing this on my VDR instance. After the fixes I mentioned in the inline comments regexps seem to be working correctly.

Thanks for testing, review and the fixes :-)

**Pull request merged**